### PR TITLE
Update workflow to store and pull streetview images from GCS subdirectory

### PR DESCRIPTION
**Pull request by nlebovits**
To better organize GCS, this PR adds a subdirectory in which to store streerview images, and also updates the front end to pull the images correctly from that directory. Addresses #717 .

### CODE REVIEW

**Comment by CodeWritingCow** [comment]
@nlebovits the "PR Checks Backend" is failing. A formatting or linting issue might be causing this. Can you take a look? Thanks!

Screen Shot 2024-09-19 at 7 19 42 PM
Screen Shot 2024-09-19 at 7 19 54 PM
Screen Shot 2024-09-19 at 7 20 22 PM

**Review by CodeWritingCow — CHANGES_REQUESTED** [review]
Hi @nlebovits this PR is running into formatting or linting issues. Can you take a look?

**Comment by nlebovits** [comment]
@CodeWritingCow yep! on it

File: data/src/streetview.py [hunk 1]
```diff
@@ -6,69 +7,148 @@
...
+failed_addresses = []
+
+
 # Helper Functions
 def get_streetview_metadata(address):
-    """Fetches metadata from the Street View API."""
-    url = f"https://maps.googleapis.com/maps/api/streetview/metadata?location={quote(address)}, Philadelphia, PA&key={key}"
-    response = requests.get(url)
-    return response.json()
+    url = f"http://example.com/metadata/{address}"
```
Comment (zigouras) [inline data/src/streetview.py:37]: This looks like an incorrect url, it used to be maps.googleapis.com

File: data/src/streetview.py [hunk 2]
```diff
@@ -6,69 +7,148 @@
...
-        print(f"Image {file_name} already exists")
-
-    else:
-        # Get streetview image
-        print(f"Fetching image for {row['address']}")
-        image_content = get_streetview_image(row["address"])
-        metadata = get_streetview_metadata(row["address"])
-        upload_image_with_metadata(blob, image_content, metadata)
+    # Check if file already exists
+    if not blob.exists():
```
Comment (zigouras) [inline data/src/streetview.py:81]: This is contrary to comment above.

File: data/src/streetview.py [hunk 3]
```diff
@@ -6,69 +7,148 @@
...
+                    blob, image_content, metadata, row["address"]
+                )
+        except Exception as e:
+            print(f"Error processing {row['address']}: {e}")
 
     time.sleep(0.5)
+
+
+# Function to retry failed addresses until all succeed
+def retry_failed_addresses():
```
Comment (zigouras) [inline data/src/streetview.py:97]: has this been tested?

File: data/src/streetview.py [hunk 4]
```diff
@@ -6,69 +7,148 @@
...
+                        process_row,
+                        [{"opa_id": None, "address": addr} for addr in failed],
+                    ),
+                    total=len(failed),
+                )
+            )
+
+
+# Retry failed addresses sequentially until
+def process_failed_addresses_sequentially():
```
Comment (zigouras) [inline data/src/streetview.py:116]: has this been tested?

**Comment by nlebovits** [comment]
@zigouras thanks for the review. there are a number of errors in the code--I took a break from working on it on Thursday but expect to get back to it this week.

**Comment by nlebovits** [comment]
@CodeWritingCow can you review the front end pieces of this commit and lmk if you see where my code is going wrong and what I can fix. No worries if not--I need to put more time into this one.

**Pull request closed without merging**